clippy_lints/src/methods/unnecessary_fallible_conversions.rs

```diff
@@ -29,6 +29,7 @@ fn check<'tcx>(
     node_args: ty::GenericArgsRef<'tcx>,
     kind: FunctionKind,
     primary_span: Span,
+    qpath: Option<&QPath<'_>>,
```

Comment:
imo this should be stored in `FunctionKind`, in `TryFromFunction` and `TryIntoFunction`, as the path is only passed in those cases (and shouldn't be for a method call)